[
  {
    "author_association": "MEMBER",
    "body": "@daxian-dbw - sorry to have just noticed this - the seemingly unnecessary `try/catch` statements did serve a useful purpose at one point. I recall hating adding them, but IIRC, it improved error reporting in some useful way. Sorry, but I don't remember more than that though.",
    "created_at": "2019-02-26T18:51:19Z",
    "html_url": "https://github.com/PowerShell/PowerShell/pull/8425#issuecomment-467564258",
    "id": 467564258,
    "issue_url": "https://api.github.com/repos/PowerShell/PowerShell/issues/8425",
    "node_id": "MDEyOklzc3VlQ29tbWVudDQ2NzU2NDI1OA==",
    "performed_via_github_app": null,
    "reactions": {
      "+1": 0,
      "-1": 0,
      "confused": 0,
      "eyes": 0,
      "heart": 0,
      "hooray": 0,
      "laugh": 0,
      "rocket": 0,
      "total_count": 0,
      "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/467564258/reactions"
    },
    "updated_at": "2019-02-26T18:51:19Z",
    "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/467564258",
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/2148248?v=4",
      "events_url": "https://api.github.com/users/lzybkr/events{/privacy}",
      "followers_url": "https://api.github.com/users/lzybkr/followers",
      "following_url": "https://api.github.com/users/lzybkr/following{/other_user}",
      "gists_url": "https://api.github.com/users/lzybkr/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/lzybkr",
      "id": 2148248,
      "login": "lzybkr",
      "node_id": "MDQ6VXNlcjIxNDgyNDg=",
      "organizations_url": "https://api.github.com/users/lzybkr/orgs",
      "received_events_url": "https://api.github.com/users/lzybkr/received_events",
      "repos_url": "https://api.github.com/users/lzybkr/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/lzybkr/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/lzybkr/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/lzybkr"
    }
  },
  {
    "author_association": "COLLABORATOR",
    "body": "Does it update ScriptStackTrace in errorrecord?",
    "created_at": "2019-02-26T19:46:43Z",
    "html_url": "https://github.com/PowerShell/PowerShell/pull/8425#issuecomment-467585501",
    "id": 467585501,
    "issue_url": "https://api.github.com/repos/PowerShell/PowerShell/issues/8425",
    "node_id": "MDEyOklzc3VlQ29tbWVudDQ2NzU4NTUwMQ==",
    "performed_via_github_app": null,
    "reactions": {
      "+1": 0,
      "-1": 0,
      "confused": 0,
      "eyes": 0,
      "heart": 0,
      "hooray": 0,
      "laugh": 0,
      "rocket": 0,
      "total_count": 0,
      "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/467585501/reactions"
    },
    "updated_at": "2019-02-26T19:46:43Z",
    "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/467585501",
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/22290914?v=4",
      "events_url": "https://api.github.com/users/iSazonov/events{/privacy}",
      "followers_url": "https://api.github.com/users/iSazonov/followers",
      "following_url": "https://api.github.com/users/iSazonov/following{/other_user}",
      "gists_url": "https://api.github.com/users/iSazonov/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/iSazonov",
      "id": 22290914,
      "login": "iSazonov",
      "node_id": "MDQ6VXNlcjIyMjkwOTE0",
      "organizations_url": "https://api.github.com/users/iSazonov/orgs",
      "received_events_url": "https://api.github.com/users/iSazonov/received_events",
      "repos_url": "https://api.github.com/users/iSazonov/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/iSazonov/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/iSazonov/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/iSazonov"
    }
  },
  {
    "author_association": "MEMBER",
    "body": "@lzybkr Thanks for bringing up that context. Looks like those try/catch were added only for functions that uses `SteppablePipeline`. Might it be related to errors thrown from within the steppable pipeline? I will play with it a bit to see if I can find out why.",
    "created_at": "2019-03-01T06:39:29Z",
    "html_url": "https://github.com/PowerShell/PowerShell/pull/8425#issuecomment-468560798",
    "id": 468560798,
    "issue_url": "https://api.github.com/repos/PowerShell/PowerShell/issues/8425",
    "node_id": "MDEyOklzc3VlQ29tbWVudDQ2ODU2MDc5OA==",
    "performed_via_github_app": null,
    "reactions": {
      "+1": 0,
      "-1": 0,
      "confused": 0,
      "eyes": 0,
      "heart": 0,
      "hooray": 0,
      "laugh": 0,
      "rocket": 0,
      "total_count": 0,
      "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/468560798/reactions"
    },
    "updated_at": "2019-03-01T06:39:29Z",
    "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/468560798",
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/127450?v=4",
      "events_url": "https://api.github.com/users/daxian-dbw/events{/privacy}",
      "followers_url": "https://api.github.com/users/daxian-dbw/followers",
      "following_url": "https://api.github.com/users/daxian-dbw/following{/other_user}",
      "gists_url": "https://api.github.com/users/daxian-dbw/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/daxian-dbw",
      "id": 127450,
      "login": "daxian-dbw",
      "node_id": "MDQ6VXNlcjEyNzQ1MA==",
      "organizations_url": "https://api.github.com/users/daxian-dbw/orgs",
      "received_events_url": "https://api.github.com/users/daxian-dbw/received_events",
      "repos_url": "https://api.github.com/users/daxian-dbw/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/daxian-dbw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/daxian-dbw/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/daxian-dbw"
    }
  },
  {
    "author_association": "MEMBER",
    "body": "The \"better error reporting\" might be related to `PowerShell.AddScript(...).Invoke()` doesn't throw on terminating error or exception, but instead keep the errors in the error stream.\r\nHere is one example of that:\r\n```\r\n$ps = [powershell]::create()\r\n$ps.AddScript(@'\r\n try {\r\n     Get-Process -Invalid 42\r\n } catch {\r\n     throw\r\n }\r\n'@)\r\n## try/catch/throw is needed for the exception to be thrown from $ps.invoke()\r\n$ps.Invoke()\r\n```\r\n\r\nBut it doesn't seem to be the exact reason for the try/catch/throw in the original `mkdir` and `oss`.\r\nThe following throws the exception even without the try/catch/throw in `mkdir`:\r\n```\r\n$ps = [powershell]::create()\r\n$ps.AddScript(\"mkdir -Path blahhfe:\\jkfjkjakfklajlkf -ea stop\") > $null\r\n$ps.Invoke()\r\n```",
    "created_at": "2019-03-02T03:56:03Z",
    "html_url": "https://github.com/PowerShell/PowerShell/pull/8425#issuecomment-468880809",
    "id": 468880809,
    "issue_url": "https://api.github.com/repos/PowerShell/PowerShell/issues/8425",
    "node_id": "MDEyOklzc3VlQ29tbWVudDQ2ODg4MDgwOQ==",
    "performed_via_github_app": null,
    "reactions": {
      "+1": 1,
      "-1": 0,
      "confused": 0,
      "eyes": 0,
      "heart": 0,
      "hooray": 0,
      "laugh": 0,
      "rocket": 0,
      "total_count": 1,
      "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/468880809/reactions"
    },
    "updated_at": "2019-03-02T03:56:03Z",
    "url": "https://api.github.com/repos/PowerShell/PowerShell/issues/comments/468880809",
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/127450?v=4",
      "events_url": "https://api.github.com/users/daxian-dbw/events{/privacy}",
      "followers_url": "https://api.github.com/users/daxian-dbw/followers",
      "following_url": "https://api.github.com/users/daxian-dbw/following{/other_user}",
      "gists_url": "https://api.github.com/users/daxian-dbw/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/daxian-dbw",
      "id": 127450,
      "login": "daxian-dbw",
      "node_id": "MDQ6VXNlcjEyNzQ1MA==",
      "organizations_url": "https://api.github.com/users/daxian-dbw/orgs",
      "received_events_url": "https://api.github.com/users/daxian-dbw/received_events",
      "repos_url": "https://api.github.com/users/daxian-dbw/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/daxian-dbw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/daxian-dbw/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/daxian-dbw"
    }
  }
]
